pkg/v1/tkg/test/tkgctl/tkgs/tkgs_workload_cluster_test.go

```diff
@@ -131,38 +135,92 @@ var _ = Describe("TKGS - Create workload cluster use cases", func() {
 		})
 		When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to true", func() {
 			BeforeEach(func() {
-				//set the cli feature flag as true -  (features.global.package-based-lcm-beta)
+				// set the cli feature flag as true -  (features.global.package-based-lcm-beta)
 				Expect(framework.SetCliConfigFlag(CLI_CLUSTERCLASS_FLAG, "true")).To(Succeed(), "error while setting CLI feature flag")
 				tkgctlClient, err = tkgctl.New(tkgctlOptions)
 				Expect(err).To(BeNil())
 			})
 			It("should return success or error based on the ClusterClass feature-gate status on the Supervisor", func() {
-				createClusterClassBasedClusterTest(tkgctlClient, deleteClusterOptions, true, clusterName, namespace)
+				createClusterClassBasedCluster(tkgctlClient, true, clusterName, namespace)
+				deleteClusterClassBasedCluster(tkgctlClient, deleteClusterOptions, clusterName, namespace)
 			})
 		})
 		When("cluster class cli feature flag (features.global.package-based-lcm-beta) is set to false", func() {
 			BeforeEach(func() {
-				//set the cli feature flag as false -  (features.global.package-based-lcm-beta)
+				// set the cli feature flag as false -  (features.global.package-based-lcm-beta)
 				Expect(framework.SetCliConfigFlag(CLI_CLUSTERCLASS_FLAG, "false")).To(Succeed(), "error while setting CLI ClusterClass flag")
 				tkgctlClient, err = tkgctl.New(tkgctlOptions)
 				Expect(err).To(BeNil())
 			})
 			It("should return success or error based on the ClusterClass feature-gate status on the Supervisor", func() {
-				createClusterClassBasedClusterTest(tkgctlClient, deleteClusterOptions, false, clusterName, namespace)
+				createClusterClassBasedCluster(tkgctlClient, false, clusterName, namespace)
```

Comment:
This whole file needs rebasing. The tests have had a big change compared to the file you have. In those Create, always happens in BeforeEach blocks and Delete happens in the AfterEach blocks. Please rebase with main

Reply:
Fixed, refactoring done in new PR